log/scope_test.go

```diff
@@ -283,13 +284,54 @@ func TestScopeErrorDictionary(t *testing.T) {
 		funcs.Store(funcs.Load().(patchTable))
 		s.WithLabels("foo", "bar").Debuga(ie, "Hello")
 
+		// Test that structured errors can be returned from a function.
+		err := func() error {
+			return structured.NewErr(ie, errors.New("err"))
+		}()
+		s.WithLabels("foo", "bar").Debugf("Hello: %s", err)
+
 		_ = Sync()
 	})
 	if err != nil {
 		t.Errorf("Got error '%v', expected success", err)
 	}
 
-	mustRegexMatchString(t, lines[0], `Hello	moreInfo=MoreInfo impact=Impact action=Action likelyCauses=LikelyCause foo=bar`)
+	mustRegexMatchString(t, lines[0], `Hello	moreInfo=MoreInfo impact=Impact action=Action likelyCause=LikelyCause 	foo=bar`)
+	mustRegexMatchString(t, lines[1], `Hello: 	moreInfo=MoreInfo impact=Impact action=Action likelyCause=LikelyCause err=err 	foo=bar`)
```

Comment:
This is still destroying the intent of `"Hello: %s"` producing a standard go error message of `context: some more context: some more context`. Especially in json output, we end up with 
```json
{
  "level": "debug",
  "time": "2020-09-17T18:46:18.149655Z",
  "scope": "TestScope",
  "msg": "",
  "message": "Hello: ",
  "moreInfo": "MoreInfo",
  "impact": "Impact",
  "action": "Action",
  "likelyCause": "LikelyCause",
  "err": "err",
  "foo": "bar"
}
```

which doesn't seem to make much sense to me

Reply:
What do you expect to see with JSON instead of the above?

Reply:
That is my main point - I don't think there *is* an output that makes sense, because I think the idea of mixing logging and `error` is not the right approach

Reply:
I'm confused... the CLI output above makes sense but there's no reasonable JSON representation of the same?
If moreInfo and the other fields were just labels and not coming from structured.Error, would the output be ok to you then?